### ISSUE TITLE
(aws_ecr_assets): DockerImageAsset cache_from not compatible with Docker < V23

### ISSUE THREAD

**Issue by BevcJ** [aws/aws-cdk#27752]
### Describe the bug

In `aws-cdk=2.83.0`, the `--cache-from` option in DockerImageAsset is generated in an incorrect format for Docker versions below v23. This conflicts with the default CodeBuild image and the available options in LinuxBuildImage, all using Docker v20, rendering the argument incompatible. Upgrading to version `aws-cdk=2.103.1` resolves the issue as the default CodeBuild image shifts to the latest version using Docker v23.

### Expected Behavior

As I understand for docker versions < 23.0 the argument should be:
```
docker build --build-arg BUILDKIT_INLINE_CACHE=1 \
--cache-from aws_account_id.dkr.ecr.region.amazonaws.com/test-cache:latest --tag aws_account_id.dkr.ecr.region.amazonaws.com/test-cache:latest .

```
Note:
- The specific image must be pulled before executing the build command.
- Setting DOCKER_BUILDKIT=1 

### Current Behavior

`--cache-from` arguments passed this way does not work.

```
docker build --build-arg BUILDKIT_INLINE_CACHE=1 \
--cache-from type=registry,ref=aws_account_id.dkr.ecr.region.amazonaws.com/test-cache:latest --tag aws_account_id.dkr.ecr.region.amazonaws.com/test-cache:latest .
```

Logs:
```
#4 importing cache manifest from type=registry
#4 sha256:93f1eeb6dd5502f7b6fb1241aec4210182a26fdda27e6428206dbb834c4cb716
#4 ERROR: invalid reference format
 
#10 importing cache manifest from ref=aws_account_id.dkr.ecr.region.amazonaws.com/test-cache:latest
#10 sha256:211a9356f59d3fa5e67b3c2ce00d0d0c859b8ecc2c83ab2fe18913486a993e6b
#10 ERROR: invalid reference format
```

### Reproduction Steps

I can provide a small example or a repository if needed for further clarification.

### CDK CLI Version

2.83.0

### Node.js Version

v18.18.2

### OS

mac

### Language

Python

**Comment by pahud** [comment]
Do you mean aws-cdk `2.103.1` has fixed this issue but you have to stick to `2.83.0`? What is your expectation?

**Comment by BevcJ** [comment]
Indeed, the compatibility seems to align from a certain version of aws-cdk onwards, specifically when latest codebuild images became available (`STANDARD_7_0` and static `AMAZON_LINUX_2_5`) and are used by default in `cdk pipelines`.

However, there appears to be an issue when utilizing the latest aws-cdk version with CodeBuild images that use Docker versions < v23 (for example `STANDARD_6_0 `).

My expectation would be to update the documentation to include information on version dependencies and the implications of using the `cache_from` argument.
